src/surrogate_heat_driver.cpp

```diff
@@ -87,40 +180,249 @@ void SurrogateHeatDriver::generate_arrays()
+  // Compute the cross-sectional areas of each region
+  solid_areas_ = xt::empty<double>({n_rings()});
+  for (gsl::index i = 0; i < n_rings(); ++i) {
+    if (i < n_fuel_rings_)
+      solid_areas_(i) = M_PI * (r_grid_fuel_(i + 1) * r_grid_fuel_(i + 1) - r_grid_fuel_(i) * r_grid_fuel_(i));
+    else {
+      int r = i - n_fuel_rings_;
+      solid_areas_(i) = M_PI * (r_grid_clad_(r + 1) * r_grid_clad_(r + 1) - r_grid_clad_(r) * r_grid_clad_(r));
+    }
+  }
+
+  // Create empty arrays for source term and temperature in the solid phase
   source_ = xt::empty<double>({n_pins_, n_axial_, n_rings()});
   temperature_ = xt::empty<double>({n_pins_, n_axial_, n_rings()});
   density_ = xt::zeros<double>({n_pins_, n_axial_, n_rings()});
   fluid_mask_ = xt::zeros<int>({n_pins_, n_axial_, n_rings()});
+
+  // Create empty arrays for temperature and density in the fluid phase
+  fluid_temperature_ = xt::empty<double>({n_pins_, n_axial_});
+  fluid_density_ = xt::empty<double>({n_pins_, n_axial_});
+}
+
+double SurrogateHeatDriver::rod_axial_node_power(const int pin, const int axial) const
+{
+  Expects(axial < n_axial_);
+
+  double power = 0.0;
+  double dz = z_(axial + 1) - z_(axial);
+
+  for (gsl::index i = 0; i < n_rings(); ++i)
+    power += source_(pin, axial, i) * solid_areas_(i) * dz;
+
+  return power;
 }
 
 void SurrogateHeatDriver::solve_step()
 {
-  std::cout << "Solving heat equation...\n";
-  // NuScale inlet temperature
-  double T_co = 523.15;
+  solve_fluid();
+  solve_heat();
+}
 
-  // Set initial temperature
-  for (auto& T : temperature_) {
-    T = T_co;
+void SurrogateHeatDriver::solve_fluid()
+{
+  // determine the power deposition in each channel; the target applications will always
+  // be steady-state or pseudo-steady-state cases with no axial conduction such that
+  // the power deposition in each channel is independent of a convective heat transfer
+  // coefficient and only depends on the rod power at that axial elevation. The channel
+  // powers are indexed by channel ID, axial ID
+  xt::xtensor<double, 2> channel_powers({n_channels_, n_axial_}, 0.0);
+  for (int i = 0; i < n_channels_; ++i) {
+    for (int j = 0; j < n_axial_; ++j) {
+      for (const auto& rod : channels_[i].rod_ids_)
+        channel_powers(i, j) += 0.25 * rod_axial_node_power(rod, j);
+    }
   }
 
+  // initial guesses for the fluid solution are uniform temperature (set to the inlet
+  // temperature) and uniform pressure  (set to the outlet pressure). These solution
+  // fields are defined on channel axial faces. The units used throughout this section
+  // are h (kJ/kg), P (MPa), u (m/s), rho (kg/m^3). Unit conversions are performed as
+  // necessary on the converged results before being used in the Monte Carlo solver.
+  // Enthalpy here requires a factor of 1e-3 to convert from J/kg to kJ/kg.
+  xt::xtensor<double, 2> h({n_channels_, n_axial_ + 1}, iapws::h1(pressure_bc_, inlet_temperature_) * 1e-3);
```

Comment:
```suggestion
  xt::xtensor<double, 2> h({n_channels_, n_axial_ + 1}, iapws::h1(pressure_bc_, inlet_temperature_));
```
I think?

Reply:
Oh that's weird, I could have sworn I tested this method out and it returned J/kg... but it looks like it's kJ/kg.